### PR TITLE
page: auth added with functionality

### PR DESCRIPTION
**Pull request by manojsarna**
- [x] Resources Added
- [x] Components Added - Header, Footer
- [x] Routes Added 
- [x] Dark Theme Added
- [x] Mockman Integration
- [x] Auth Page Added

###Check these Files 
[auth-context.js](https://github.com/manojsarna/manoj-smasher-socialmedia-react/compare/auth-functionality?expand=1#diff-2ff9fb4b317b5c2dccb4268b6b97be89585b20d422a11254c3d590ec13d83a77)

[Auth.jsx](https://github.com/manojsarna/manoj-smasher-socialmedia-react/compare/auth-functionality?expand=1#diff-05a2ef413bbfcc66caca4c90c620ada11233c10128abf287208bfdeddc429ba8)

### Ignore Rest of the Files

image

### CODE REVIEW

File: src/pages/auth/Auth.jsx [hunk 1]
```diff
@@ -0,0 +1,19 @@
+import { useState } from "react";
+import { useDocTitle } from "../../hooks/useDocTitle";
```
Comment (pankaj-prog) [inline src/pages/auth/Auth.jsx:2]: suggestion: use absolute imports

File: src/pages/auth/auth.css [hunk 2]
```diff
@@ -0,0 +1,187 @@
...
+.sm-main-auth-login,
+.sm-main-auth-signup {
+  display: flex;
+  flex-direction: column;
+  justify-content: center;
+  align-items: center;
+  width: fit-content;
+  gap: 1rem;
+  padding: 1rem 6rem;
+  padding-bottom: 3rem;
```
Comment (pankaj-prog) [inline src/pages/auth/auth.css:22]: you can write it in one line

File: src/pages/auth/auth.css [hunk 3]
```diff
@@ -0,0 +1,187 @@
...
+
+.sm-btn-text {
+  background: none;
+  border: none;
+  cursor: pointer;
+  font-size: 1.6rem;
+  padding: 1rem 1.5rem;
+  font-weight: 700;
+  color: var(--COLOR-BLACK);
+  /* transition: all ease-in 0.1s; */
```
Comment (pankaj-prog) [inline src/pages/auth/auth.css:78]: remove comment

File: src/pages/auth/components/Login.jsx [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+    } else {
+      setErrorMessage({
+        show: true,
+        message: "Invalid Credentials",
+      });
+    }
+  };
+
+  return (
+    <div id="auth-login" className="sm-main-auth-login">
```
Comment (pankaj-prog) [inline src/pages/auth/components/Login.jsx:35]: you can use form here, better for accesibility

**Review by pankaj-prog — APPROVED** [review]

File: src/pages/auth/Auth.jsx (hunk 1 above)
Comment (tarunsankhla) [inline src/pages/auth/Auth.jsx:2]: You can use jscomfig and add absolute path

File: src/redux/reducers/authSlice.js [hunk 5]
```diff
@@ -0,0 +1,109 @@
...
+export const logout = createAsyncThunk("auth/logout", async () => {
+  localStorage.removeItem("smasherToken");
+});
+
+const authSlice = createSlice({
+  name: "auth",
+  initialState,
+  extraReducers(builder) {
+    builder
+      .addCase(login.fulfilled, (state, action) => {
```
Comment (tarunsankhla) [inline src/redux/reducers/authSlice.js:71]: Appreciation on use of redux

File: src/pages/user/User.jsx [hunk 6]
```diff
@@ -1,16 +1,20 @@
-import { useAuth } from "../../context";
+import { useDispatch, useSelector } from "react-redux";
+
 import { useDocTitle } from "../../hooks/useDocTitle";
```
Comment (tarunsankhla) [inline src/pages/user/User.jsx:3]: Remove extra space

**Review by tarunsankhla — APPROVED** [review]
The code looks great, preview deployed not available

**Pull request merged**